app/useCases/getAsteroidData.js

```diff
@@ -1,14 +1,36 @@
-﻿const { fetchAsteroidDataForWeekDates } = require('../repository/nasaApi');
-const { formatAsteroidData, getWeekDates } = require('../utils/utils');
+﻿const {fetchAsteroidData} = require('../repository/nasaApi');
+const {formatAsteroidData} = require('../utils/utils');
 
-async function getAsteroidData(apiUrl, apiKey) {
-    const {startDate, endDate} = getWeekDates();
-    const data = await fetchAsteroidDataForWeekDates(apiUrl, apiKey, startDate, endDate);
+async function getAsteroidData(apiUrl, apiKey, startDate, endDate, includeSentryObjectsCount, checkDangerousMeteors) {
+    const data = await fetchAsteroidData(apiUrl, apiKey, startDate, endDate);
     const asteroids = data.near_earth_objects;
 
-    return Object.keys(asteroids)
-        .map(date => formatAsteroidData(asteroids[date]))
-        .flat();
+    let formattedData = Object.keys(asteroids)
+        .flatMap(date => formatAsteroidData(asteroids[date]));
+
+    let response = { data: formattedData };
+
+    if (checkDangerousMeteors) {
+        const hasDangerousMeteors = Object.keys(asteroids)
+            .flatMap(date => asteroids[date])
+            .some(asteroid => asteroid.is_potentially_hazardous_asteroid);
+
+        if (hasDangerousMeteors) {
+            response.wereDangerousMeteors = true;
+        }
+    }
+
+    if (includeSentryObjectsCount) {
+        let sentryAsteroids = Object.keys(asteroids)
+            .flatMap(date => asteroids[date])
+            .filter(asteroid => asteroid.is_sentry_object);
+
+        if (sentryAsteroids.length > 0) {
```

Comment:
you can check that the array is not empty by simply using `if(sentryAsteroids.length)`, no necessary to add `>0`